scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1286,6 +1312,67 @@ def _create_build_dir_link(self, links_dir_path, instance):
 
         self.link_dir_counter += 1
 
+    def handle_additional_scripts(
+        self, platform_name: str, testsuite: TestInstance
+    ) -> bool:
+        logger.debug(testsuite.testsuite.id)
+        matched_scripting = self.scripting.get_matched_scripting(
+            testsuite.testsuite.id, platform_name
+        )
+        if matched_scripting:
+            # Define a function to validate
+            # if the platform is supported by the matched scripting
+            def validate_boards(platform_scope, platform_from_yaml):
+                return any(board in platform_scope for board in platform_from_yaml)
```

Comment:
looks like this check can be called only once, right ?
```suggestion
            if not any(board in platform_name for board in matched_scripting.platforms):
                return False
```